x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/actions/ActionWrapperTests.java

```diff
@@ -193,11 +193,48 @@ public void testLimitOfNumberOfActionsExecuted() throws Exception {
             assertThat(map.get("foreach"), instanceOf(List.class));
             List<Map<String, Object>> actions = (List) map.get("foreach");
             assertThat(actions, hasSize(100));
+            assertThat(map, hasKey("max_iterations"));
+            assertThat(map.get("max_iterations"), is(100));
             assertThat(map, hasKey("number_of_actions_executed"));
             assertThat(map.get("number_of_actions_executed"), is(100));
         }
     }
 
+    public void testConfiguredLimitOfNumberOfActionsExecuted() throws Exception {
```

Comment:
If I understand correctly this test should not need to changes to any assertions if your max_iterations is 1 or 50 or 1000.  Could you grab a random value between 1 and 1000 (org.elasticsearch.test.ESTestCase#randomIntBetween) ?  and use that instead of 50 ?

Reply:
Thanks for the feedback @jakelandis, 50 was just an arbitrary number. I'll be pushing the changes soon 🙂